src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -248,31 +248,31 @@ static WithTypeParametersBinder getExtraWithTypeParametersBinder(Binder enclosin
 
             // We're bringing parameters in scope inside `nameof` in attributes on methods, their type parameters and parameters.
             // This also applies to local functions, lambdas, indexers and delegates.
-            static InMethodBinder getExtraInMethodBinder(Binder enclosing, Symbol target)
+            static WithParametersBinder getExtraWithParametersBinder(Binder enclosing, Symbol target)
             {
-                var method = target switch
+                var parameters = target switch
                 {
-                    MethodSymbol methodSymbol => methodSymbol,
-                    ParameterSymbol parameter => getMethodFromParameter(parameter),
-                    TypeParameterSymbol typeParameter => getMethodFromTypeParameter(typeParameter),
-                    PropertySymbol property => property.GetMethod,
-                    NamedTypeSymbol namedType when namedType.IsDelegateType() => namedType.DelegateInvokeMethod,
-                    _ => null
+                    MethodSymbol methodSymbol => methodSymbol.Parameters,
+                    ParameterSymbol parameter => getAllParameters(parameter),
+                    TypeParameterSymbol typeParameter => getMethodFromTypeParameter(typeParameter).Parameters,
+                    PropertySymbol property => property.Parameters,
+                    NamedTypeSymbol namedType when namedType.IsDelegateType() => namedType.DelegateInvokeMethod.Parameters,
+                    _ => default
                 };
 
-                return method is null
+                return parameters.IsDefaultOrEmpty
                     ? null
-                    : new InMethodBinder(method, enclosing);
+                    : new WithParametersBinder(parameters, enclosing);
             }
 
-            static MethodSymbol getMethodFromParameter(ParameterSymbol parameter)
+            static ImmutableArray<ParameterSymbol> getAllParameters(ParameterSymbol parameter)
             {
                 var containingSymbol = parameter.ContainingSymbol;
                 return containingSymbol switch
                 {
-                    MethodSymbol method => method,
-                    PropertySymbol property => property.GetMethod,
-                    _ => null
+                    MethodSymbol method => method.Parameters,
+                    PropertySymbol property => property.Parameters,
```

Comment:
Do we expect to be able to do `nameof(value)` when an attribute is applied to a property setter?

Reply:
Good question. I blocked the scope change on accessors and started an email thread to clarify expectations.